pennylane/_qubit_device.py

```diff
@@ -380,18 +380,47 @@ def execute_new(self, circuit, **kwargs):
         # apply all circuit operations
         self.apply(circuit.operations, rotations=circuit.diagonalizing_gates, **kwargs)
 
-        results = self.statistics(circuit.observables)
+        # generate computational basis samples
+        if self.shots is not None or circuit.is_sampled:
```

Comment:
```suggestion
        if self.shots is not None
```
Probably unnecessary because if the circuit is sampled then the shots are finite as well